src/shared/components/TextExpander.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {observer} from "mobx-react";
+import {If, Else, Then} from 'react-if';
+import TextTruncate from 'react-text-truncate';
+import {observable} from "mobx";
+
+export interface ITextExpanderProps
+{
+    text?: string|JSX.Element|null;
+    line?: number;
+    expandIndicator?: string|JSX.Element;
+    collapseIndicator?: string|JSX.Element;
+    truncateText?: string;
+}
+
+@observer
+export default class TextExpander extends React.Component<ITextExpanderProps, {}> {
+    @observable protected isTextTruncated: boolean = true;
```

Comment:
interesting.  here we are really replacing React's setState with mobx, even internally. i'm not saying this a bad decision, but elsewhere we are using state.  we should perhaps discuss what our recommendation is.